.circleci/Dockerfile

```diff
@@ -10,19 +10,20 @@ FROM circleci/openjdk:latest
 
 # clang is used for TSAN and ASAN tests
 RUN sudo sh -c 'curl http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -'
-RUN sudo sh -c 'echo "deb http://apt.llvm.org/stretch/ llvm-toolchain-stretch-7 main" > /etc/apt/sources.list.d/llvm.list'
+RUN sudo sh -c 'echo "deb http://apt.llvm.org/stretch/ llvm-toolchain-stretch-8 main" > /etc/apt/sources.list.d/llvm.list'
 
 RUN sudo apt-get update && \
     sudo apt-get -y install \
     wget software-properties-common make python python-pip pkg-config \
     zlib1g-dev bash-completion bc libtool automake zip time g++-6 gcc-6 \
-    clang-7 clang-format-7 clang-tidy-7 lld-7 libc++-7-dev libc++abi-7-dev \
+    clang-8 clang-format-8 clang-tidy-8 lld-8 libc++-8-dev libc++abi-8-dev \
     rsync ninja-build
 
-# ~100M, depends on g++, zlib1g-dev, bash-completions
-RUN curl -Lo /tmp/bazel.deb https://github.com/bazelbuild/bazel/releases/download/0.25.0/bazel_0.25.0-linux-x86_64.deb && \
-    sudo dpkg -i /tmp/bazel.deb && rm /tmp/bazel.deb
-
+# Use Bazelisk launcher for Bazel (automatically downloads required version of Bazel).
+RUN cd /tmp && \
```

Comment:
being Dockerfile pedantic here, but usually, it is good practice to avoid cd  and use WORKDIR
c.f. https://github.com/hadolint/hadolint/wiki/DL3003

I'm seeing that it is used in the following RUN command (which actually makes a stronger point of placing a WORKDIR /tmp before this 2 RUN), but it can be refactored in the future.

Reply:
Patches welcomed, but that's merged from `master`, not it's not unique to this PR / branch.